receiver/gitproviderreceiver/config.go

```diff
@@ -44,7 +44,7 @@ func (cfg *Config) Unmarshal(componentParser *confmap.Conf) error {
 	}
 
 	// load the non-dynamic config normally
-	err := componentParser.Unmarshal(cfg)
+	err := componentParser.Unmarshal(cfg, confmap.WithIgnoreUnused())
```

Comment:
Migrating here to make it explicit - the default behavior is now to error out on unused elements.